coalib/bearlib/aspects/Smell.py

```diff
@@ -0,0 +1,313 @@
+from coalib.bearlib.aspects import Taste, Root
+
+
+@Root.subaspect
+class Smell:
+    """
+    This aspect detects `code smells` or `bad smells` in your code.
+
+    `Smells` are certain structures in the code that indicate violation of
+    fundamental design principles. They are usually not bugs; they are not
+    technically incorrect and do not currently prevent the program from
+    functioning.
+    """
+    class docs:
+        example = """
+        * Feature envy
+        * Data clump
+        * Too large class
+        * Too long parameter list
+        etc...
+        """
+        example_language = 'English'
+        importance_reason = """
+        Even though they are not necessarily bugs, code smells increase the risk
+        of bugs or failure in the future and may slow down development.
+        """
+        fix_suggestions = 'some fix suggestions'
+
+
+@Smell.subaspect
+class ClassSmell:
+    """
+    This aspect detects `code smells` or `bad smells` related to classes'
+    definitions in your codebase.
+
+    Class-level code smells are simply code smells indicating poorly defined
+    classes (including too large classes or God object, data clump feature
+    envy etc...) in our source code.
+    """
+    class docs:
+        example = """
```

Comment:
Examples should be clearer. I mean if you can provide code samples it would be great.